XL build failures with Kokkos::complex<Kokkos::Experimental::half_t> starting on 10-01-21

One of the kokkos-kernels nightly builds started failing with:
```
kokkos-install/include/Kokkos_Complex.hpp:80:1: error: exception specification of explicitly defaulted default constructor does not match the calculated one
kokkos-kernels/example/gmres/gmres.hpp:62:49: note: in instantiation of template class 'Kokkos::complex<Kokkos::Experimental::half_t>' requested here
07:24:52 Kokkos::complex< Kokkos::Experimental::half_t>  abs(Kokkos::complex< Kokkos::Experimental::half_t>  arg) noexcept { 
```

The following reproducer from @ndellingwood can be used to reproduce the error:
Tested on weaver testbed (Power9+Volta70 machine)

### Small code test:
```c++
#include <iostream>
#include "Kokkos_Core.hpp"

int main()
{
  std::cout << "Begin code" << std::endl;
  Kokkos::ScopeGuard();
  {
    Kokkos::complex<double> kcd;
    Kokkos::complex<Kokkos::Experimental::half_t> kcht;
  }
  return 0;
}
```

### Makefile, hacked from the kokkos-tutorials exercises to use xlC as the host compiler:
```
KOKKOS_PATH = ${KOKKOSPATH}
KOKKOS_DEVICES = "Cuda,Serial"
EXE_NAME = "TestComplexHalft"

SRC = $(wildcard *.cpp)

default: build
        echo "Start Build"


ifneq (,$(findstring Cuda,$(KOKKOS_DEVICES)))
CXX = ${KOKKOS_PATH}/bin/nvcc_wrapper
EXE = ${EXE_NAME}.cuda
KOKKOS_ARCH = "Power9,Volta70"
KOKKOS_CUDA_OPTIONS = "enable_lambda,force_uvm"
else
CXX = g++
EXE = ${EXE_NAME}.host
endif

CXXFLAGS = -qxflag=disable__cplusplusOverride, -ccbin xlC
LINK = ${CXX}
LINKFLAGS =

DEPFLAGS = -M

OBJ = $(SRC:.cpp=.o)
LIB =

include $(KOKKOS_PATH)/Makefile.kokkos

build: $(EXE)

$(EXE): $(OBJ) $(KOKKOS_LINK_DEPENDS)
        $(LINK) $(KOKKOS_LDFLAGS) $(LINKFLAGS) $(EXTRA_PATH) $(OBJ) $(KOKKOS_LIBS) $(LIB) -o $(EXE)

clean: kokkos-clean
        rm -f *.o *.cuda *.host

# Compilation rules

%.o:%.cpp $(KOKKOS_CPP_DEPENDS)
        $(CXX) $(KOKKOS_CPPFLAGS) $(KOKKOS_CXXFLAGS) $(CXXFLAGS) $(EXTRA_INC) -c $<

test: $(EXE)
        ./$(EXE)
```
### Environment:
```
module purge
module load cuda/10.1.105 gcc/7.4.0 ibm/xl/16.1.1
```
### Error output:
```
TestComplexHalft.cpp(10): error: the default constructor of "Kokkos::complex<Kokkos::Experimental::half_t>" cannot be referenced -- it is a deleted function
```

--- a/core/src/Kokkos_Complex.hpp
+++ b/core/src/Kokkos_Complex.hpp
@@ -77,7 +77,7 @@ class
 
   //! Default constructor (initializes both real and imaginary parts to zero).
   KOKKOS_DEFAULTED_FUNCTION
-  complex() noexcept = default;
+  complex() = default;
 
   //! Copy constructor.
   KOKKOS_DEFAULTED_FUNCTION

